data/src/message.rs

```diff
@@ -173,19 +191,168 @@ pub enum Direction {
     Received,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct Reaction {
+    pub target: Target,
+    pub sender: Nick,
+    pub text: String,
+    pub id: Option<Id>,
+    pub in_reply_to: Id,
+    pub unreact: bool,
+}
```

Comment:
We should have a different representation of a reaction for how its identified vs how its stored. When storing it, we only care about which message id it relates to. We don't need to store `unreact` reactions because we're simply removing them from history metadata. Same with `target`, this is only needed to figure out which history buffer to apply the reaction to, but once stored, its implied by where its stored. 

It might be good to have a  `reaction` module that allows us to describe both forms, one used at runtime for routing from decoding -> history. And one that we actually store in history that's slim.